src/plugins/action_generate_review_assignment_records/src/generateReviewAssignments.ts

```diff
@@ -116,6 +109,111 @@ module.exports['generateReviewAssignments'] = async function (input: any, DBConn
+  // TODO: Needs to check if there are already reviews created - if partial review
+  // for level - 1 are submitted separetedily will create 2 reviewAssignments for level 1++
+  const nextLevelReviewers = await db.getPersonnelForApplicationStageLevel(
+    templateId,
+    nextStageNumber,
+    nextReviewLevel,
+    'Review'
+  )
+  console.log('Next level reviewers', nextLevelReviewers)
+  const reviewAssignments: ReviewAssignmentObject = {}
+
+  // Build reviewers into object map so we can combine duplicate user_orgs
+  // and merge their section code restrictions
+  nextLevelReviewers.forEach((reviewer: Reviewer) => {
+    const { userId, orgId, restrictions } = reviewer
+
+    const templateSectionRestrictions = restrictions
+      ? restrictions?.templateSectionRestrictions
+      : null
+
+    const userOrgKey = `${userId}_${orgId ? orgId : 0}`
+    if (reviewAssignments[userOrgKey])
+      reviewAssignments[userOrgKey].templateSectionRestrictions = mergeSectionRestrictions(
+        reviewAssignments[userOrgKey].templateSectionRestrictions,
+        templateSectionRestrictions
+      )
+    else
+      reviewAssignments[userOrgKey] = {
+        reviewerId: userId,
+        orgId,
+        stageId: nextStageId,
+        stageNumber: nextStageNumber,
+        // TO-DO: allow STATUS to be configurable in template
+        status: restrictions?.canSelfAssign
+          ? AssignmentStatus.SELF_ASSIGN
+          : nextReviewLevel === 1
+          ? AssignmentStatus.AVAILABLE
+          : AssignmentStatus.SELF_ASSIGN,
+        applicationId,
+        templateSectionRestrictions,
+        level: nextReviewLevel,
+        isLastLevel,
+      }
+  })
+  // Save review_assignment records to database
+  const reviewAssignmentIds = await db.addReviewAssignments(Object.values(reviewAssignments))
+
+  // Generate review_assignment_assigner_joins
+  // For now we assume that assigners have no Section restrictions
+  console.log('Generating review_assignment_assigner_join records...')
+  const availableAssigners = await db.getPersonnelForApplicationStageLevel(
+    templateId,
+    nextStageNumber,
+    nextReviewLevel,
+    'Assign'
+  )
+  const reviewAssignmentAssignerJoins = []
+  for (const reviewAssignmentId of reviewAssignmentIds) {
+    for (const assigner of availableAssigners) {
+      reviewAssignmentAssignerJoins.push({
+        assignerId: assigner.userId,
+        orgId: assigner.orgId,
+        reviewAssignmentId,
+      })
+    }
+  }
+  const reviewAssignmentAssignerJoinIds = await db.addReviewAssignmentAssignerJoins(
+    reviewAssignmentAssignerJoins
+  )
+
+  console.log('ReviewAssignmentAssignerJoinIds', reviewAssignmentAssignerJoinIds)
+
+  return {
+    status: 'Success',
+    error_log: '',
+    output: {
+      reviewAssignments: Object.values(reviewAssignments),
+      reviewAssignmentIds,
+      reviewAssignmentAssignerJoins,
+      reviewAssignmentAssignerJoinIds,
+      nextStageNumber,
+      nextReviewLevel,
+    },
+  }
+}
```

Comment:
Is this all the same as it was, just extracted into a method?

Reply:
Yes